cmdstanpy/stanfit.py

```diff
@@ -754,20 +754,23 @@ def stan_variable(self, name: str) -> np.ndarray:
         dims = self._stan_variable_dims[name]
         if dims == 1:
             idx = self.column_names.index(name)
-            return self._draws[self._draws_warmup :, :, idx].reshape(
-                (dim0,), order='A'
-            )
+            return pd.DataFrame({
+                name: self._draws[self._draws_warmup:, :, idx].reshape(
+                    (dim0,), order='A'
+                )
+            })
         else:
             idxs = [
-                x[0]
+                x
                 for x in enumerate(self.column_names)
                 if x[1].startswith(name + '.')
             ]
-            var_dims = [dim0]
-            var_dims.extend(dims)
-            return self._draws[
-                self._draws_warmup :, :, idxs[0] : idxs[-1] + 1
-            ].reshape(tuple(var_dims), order='A')
+            return pd.DataFrame({
+                n: self._draws[
+                    self._draws_warmup:, :, x
+                ].reshape(dim0, order='A')
+                for x, n in idxs
+            })
```

Comment:
Hi, I think there is now a possibility to streamline this a bit, at the same time we can use regex to find columns.

Please check if this code works.

Import re at start
```
import re
```

This works for all the variable types. Also it takes one chunk on transforms that to dataframe, so no need for dictionary handling.


```
dims = self._stan_variable_dims[name]
idxs = []
names = []
pattern = r'^{}(\.\d)*$'.format(name)
for i, column_name in enumerate(self.column_names):
    if re.search(pattern, column_name):
        names.append(column_name)
        idxs.append(i)

var_dims = [dim0]
var_dims.extend(dims)
return pd.DataFrame(self._draws[
    self._draws_warmup :, :, idxs[0] : idxs[-1] + 1
].reshape(tuple(var_dims), order='A'), columns=names)
```

Reply:
Hey! 
Thanks a lot! Of course you're right. I made a few tweaks to your suggestion to make it work. Hope that's fine.